weave/trace_server/trace_server_interface.py

```diff
@@ -318,6 +319,75 @@ class FileContentReadRes(BaseModel):
     content: bytes
 
 
+class FeedbackPayloadReactionReq(BaseModel):
+    emoji: str
+
+
+class FeedbackPayloadNoteReq(BaseModel):
+    note: str = Field(min_length=1, max_length=1024)
+
+
+class FeedbackCreateReq(BaseModel):
+    project_id: str = Field(examples=["entity/project"])
+    weave_ref: str = Field(examples=["weave:///entity/project/object/name:digest"])
+    creator: typing.Optional[str] = Field(default=None, examples=["Jane Smith"])
+    feedback_type: str = Field(examples=["custom"])
+    payload: typing.Dict[str, typing.Any] = Field(
+        examples=[
+            {
+                "key": "value",
+            }
+        ]
+    )
+
+
+class FeedbackCreateReqForInsert(FeedbackCreateReq):
+    wb_user_id: str
+
+
+# The response provides the additional fields needed to convert a request
+# into a complete Feedback.
+class FeedbackCreateRes(BaseModel):
+    id: str
+    created_at: datetime.datetime
+    wb_user_id: str
+    payload: typing.Dict[str, typing.Any]  # If not empty, replace payload
+
+
+class Feedback(FeedbackCreateReqForInsert):
+    id: str
+    created_at: datetime.datetime
+
+
+class FeedbackQueryReq(BaseModel):
+    project_id: str = Field(examples=["entity/project"])
+    fields: typing.Optional[list[str]] = Field(
+        default=None, examples=[["id", "feedback_type", "payload.note", "count(*)"]]
```

Comment:
interesting idea to combine `count(*)` in the field selection. I'm curious to read more below how this is handled - both from a sql injection perspective but also if you have count & fields, then you need 2 queries

Reply:
I haven't added `GROUP BY` or `HAVING` to ORM yet. Kind of hacked in `count(*)` because it was useful for tests. Re: injection, I'm only allowing that specific string, we'd have to get fancier if we wanted to support functions more generally.

Reply:
if it is just useful for tests, i might not advertise it as a primary example in case you want to change the way aggregations are specified when you are doing an agg operation